### PR TITLE
[nanobench] Bump to 4.3.10

### PR DESCRIPTION
**Pull request by xiaozhuai**
1. Bump to 4.3.10
2. Remove dependencies `fmt`, not needed at all.
3. Modernization

#### Why provide a CMakeLists.txt instead of using the project's own.
The project's own CMakeLists.txt is not well performed.
Rewrite it from scratch is better than patch it since it's a header-only library.
It also makes maintaining the port simpler.
And it follows the official's export names.

### CODE REVIEW

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

File: ports/nanobench/portfile.cmake [hunk 1]
```diff
@@ -1,22 +1,25 @@
 vcpkg_minimum_required(VERSION 2022-11-10)
```
[the remark this answers is not in the document]
Comment (xiaozhuai) [reply ports/nanobench/portfile.cmake]: Done

**Review by LilyWangLL — APPROVED** [review]

**Comment by RT222** [comment]
I had undefined symbols error after I upgraded to this PR. For people having the same issue, in order to fix it you need to define the ANKERL_NANOBENCH_IMPLEMENT macro before including nanobench.h.

Also, I disagree with the changes made in this PR. The nanobench header is split in two parts : the declaration and the implementation. The later is disabled by default, and must be enabled with the ANKERL_NANOBENCH_IMPLEMENT macro if one want to use the library as single header.

I can also be built as a library (with the ANKERL_NANOBENCH_IMPLEMENT enabled), and the nanobench header can then be included without the macro to bring only the declaration part. This is the default behavior, and this is better in my opinion, because you don't need to define the macro every time you want to include the header, and more importantly because it improves the compilation speed.

**Comment by xiaozhuai** [comment]
> I had undefined symbols error after I upgraded to this PR. For people having the same issue, in order to fix it you need to define the ANKERL_NANOBENCH_IMPLEMENT macro before including nanobench.h.
> 
> Also, I disagree with the changes made in this PR. The nanobench header is split in two parts : the declaration and the implementation. The later is disabled by default, and must be enabled with the ANKERL_NANOBENCH_IMPLEMENT macro if one want to use the library as single header.
> 
> I can also be built as a library (with the ANKERL_NANOBENCH_IMPLEMENT enabled), and the nanobench header can then be included without the macro to bring only the declaration part. This is the default behavior, and this is better in my opinion, because you don't need to define the macro every time you want to include the header, and more importantly because it improves the compilation speed.

Hi, @RT222 , thanks for you feedback, I'll submit a PR to revert to the default behavior.

**Comment by RT222** [comment]
Thanks for this and for the update to the latest version. :)

**Comment by xiaozhuai** [comment]
> Thanks for this and for the update to the latest version. :)

How about this?
```cmake
find_package(nanobench CONFIG REQUIRED)

# will link to nanobench library
target_link_libraries(main PRIVATE nanobench::nanobench)

# this require "#define ANKERL_NANOBENCH_IMPLEMENT" before #include <nanobench.h>
target_link_libraries(main PRIVATE unofficial::nanobench::nanobench_headeronly)
```

**Comment by RT222** [comment]
If I got it right, you would still have to build and provide the library file to allow the user to choose? If this is the case, I fail to see the benefit of using the header only version. Why do you need to build this library as header only? Are you only trying to simplify the port?

**Comment by xiaozhuai** [comment]
> If I got it right, you would still have to build and provide the library file to allow the user to choose? If this is the case, I fail to see the benefit of using the header only version. Why do you need to build this library as header only? Are you only trying to simplify the port?

No, I will update this port, it will provide two verison. If you link to `nanobench::nanobench` (the default), it do not need `#define ANKERL_NANOBENCH_IMPLEMENT`
Otherwise if you link to `unofficial::nanobench::nanobench_headeronly`, it need to `#define ANKERL_NANOBENCH_IMPLEMENT` before including.

Two choice.

**Pull request merged**